### PR TITLE
improvements in func-test, enable server42

### PR DESCRIPTION
**Pull request by rtretyak**
- discovery packet ttl increased to allow bigger delay to switches
- otsdb client noisy output fixed
- enable server42 by default
- adjust functional to work with server42 rules in multiTable mode
- refactor getDefaultCookies according to #3853
- unignore functional test for 3821

### CODE REVIEW

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/HealthCheckSpecification.groovy [hunk 1]
```diff
@@ -10,8 +10,13 @@ import org.openkilda.messaging.info.event.IslChangeType
 import org.openkilda.testing.model.topology.TopologyDefinition.Status
 import org.openkilda.testing.tools.SoftAssertions
 
+import org.springframework.beans.factory.annotation.Value
+
 class HealthCheckSpecification extends BaseSpecification {
 
+    @Value('${use.multitable}')
+    boolean useMultitable
```
Comment (andriidovhan) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/HealthCheckSpecification.groovy:18]: How about moving it to the `BaseSpecification` file?
After that, we will refactor other specs and delete defining of the `useMultitable` variable

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/HealthCheckSpecification.groovy [hunk 2]
```diff
@@ -64,5 +69,27 @@ class HealthCheckSpecification extends BaseSpecification {
...
+            server42FlowRtt
+            //for below props any value allowed. dependent tests will skip themselves or adjust if feature is off
+            //flowsRerouteUsingDefaultEncapType
+            //collectGrpcStats
+        }
+
+        and: "Switches configurations are in expected state"
+        topology.switches.each { sw ->
+            verifyAll(northbound.getSwitchProperties(sw.dpId)) {
+                multiTable == useMultitable
```
Comment (andriidovhan) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/HealthCheckSpecification.groovy:90]: It won't work for hardware env.
Some switch doesn't support the `multiTable` mode
```
topology.activeSwitches.each {sw ->
            if (sw.features.contains(SwitchFeature.MULTI_TABLE)) {
                verifyAll(northbound.getSwitchProperties(sw.dpId)) {
                    multiTable == useMultitable
                }
            }
        }
```

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/resilience/RetriesSpec.groovy [hunk 3]
```diff
@@ -32,7 +32,7 @@ import java.util.concurrent.TimeUnit
 
 @Slf4j
 class RetriesSpec extends HealthCheckSpecification {
-    @Shared int globalTimeout = 30
+    @Shared int globalTimeout = 30 //global itmeout for h&s operation
```
Comment (andriidovhan) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/resilience/RetriesSpec.groovy:35]: `itmeout -> timeout`

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/HealthCheckSpecification.groovy (hunk 2 above)
Comment (rtretyak) [reply src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/HealthCheckSpecification.groovy:90]: changed to
```
            verifyAll(northbound.getSwitchProperties(sw.dpId)) {
                multiTable == useMultitable && sw.features.contains(SwitchFeature.MULTI_TABLE)
```

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/HealthCheckSpecification.groovy (hunk 1 above)
Comment (rtretyak) [reply src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/HealthCheckSpecification.groovy:18]: done, also removed duplicates from child specs

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/resilience/RetriesSpec.groovy (hunk 3 above)
Comment (rtretyak) [reply src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/resilience/RetriesSpec.groovy:35]: done

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/server42/Server42RttSpec.groovy [hunk 4]
```diff
@@ -1,96 +1,96 @@
...
-        when: "Set server42FlowRtt toggle to false"
-        def flowRttFeatureStartState = isEnabledFlowRtt();
-        disableFlowRtt()
+        and: "server42FlowRtt toggle is turned off"
+        def flowRttFeatureStartState = changeFlowRttToggle(false)
 
-        and: "Set server42Switch FlowRtt deactivated"
-        def flowRttSwitchFeatureStartState = isEnabledFlowRttForSwitch(server42Switch);
-        disableFlowRttForSwitch(server42Switch)
+        and: "server42FlowRtt for src switch is turned off on src and dst"
```
Comment (andriidovhan) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/server42/Server42RttSpec.groovy:92]: looks like it should be paraphrased
probably to: ` and: "server42FlowRtt is turned off on src and dst"`

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/server42/Server42RttSpec.groovy [hunk 5]
```diff
@@ -1,96 +1,96 @@
...
+ */
 class Server42RttSpec extends HealthCheckSpecification {
     @Shared
     @Value('${opentsdb.metric.prefix}')
     String metricPrefix
 
-    def "Create two flow with server42 Rtt feature and check datapoints in opentsdb"() {
-        given: "Two active neighboring switches one with server42 and one without"
+    @Tidy
+    def "Flow rtt stats are available in forward and reverse directions for new flows"() {
```
Comment (andriidovhan) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/server42/Server42RttSpec.groovy:32]: what do you mean `for new flows`? Flow rtt stats are not available for a flow that is created before enabling server42FlowRtt , am I right?

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/resilience/RetriesSpec.groovy [hunk 6]
```diff
@@ -412,9 +412,7 @@ and at least 1 path must remain safe"
         antiflap.portDown(islToBreak.srcSwitch.dpId, islToBreak.srcPort)
 
         and: "Connection to src switch is slow in order to simulate a global timeout on reroute operation"
-        lockKeeper.shapeSwitchesTraffic([swPair.src], new TrafficControlData(5000))
-        //note that for some reason 5000+ delay will also cause isls on given switch to timeout. Reason unknown
-        //failed isls have no impact on this particular scenario
+        lockKeeper.shapeSwitchesTraffic([swPair.src], new TrafficControlData(7000))
```
[the remark this answers is not in the document]
Comment (rtretyak) [reply src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/resilience/RetriesSpec.groovy:415]: more stability. 5000 fails in some cases due to no global timeout happening

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/server42/Server42RttSpec.groovy (hunk 5 above)
Comment (rtretyak) [reply src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/server42/Server42RttSpec.groovy:32]: This text is not implying that it is not available for old flows, it is only stating what is checked in the test - rtt is available for new flows (aka flows created after enabling toggles)

**Review by andriidovhan — APPROVED** [review]

**Pull request merged**